src/dask_awkward/lib/optimize.py

```diff
@@ -42,6 +42,10 @@ def all_optimizations(
     else:
         # Perform dask-awkward specific optimizations.
         dsk = optimize(dsk, keys=keys)
+
+        # blockwise layer chaining optimization.
```

Comment:
this should go in `def optimize` below so it gets called when dask_histogram is the final layer.